elastic/kibana
# [Mappings editor] Remove boost parameter from field types

This PR removes support for the `boost` parameter in the mappings editor from `8.x`.

## Updated types

- `boolean`
- `date` and `date_nanos`
- `flattened`
- `ip`
- `keyword`
- `numeric` (all of them)
- `range` (all of them)
- `text`
- `token_count`

I am going to check if index templates or component templates with mappings that contains the `boost` parameters are surfaced in Upgrade assistant. As this should be done ES side it does not block merging this work.

## How to test

* Navigate to index management > Index template tab
* Create a new template
* Fill the logistic step and navigate to the mappings editor
* Create a field
* Edit the field and make sure that the "boost" parameter is not present anymore
* Change the type to one of the list above and make sure that the boost parameter not present

Fixes https://github.com/elastic/kibana/issues/78293
